Task 4 employees table

Alenapadsekina opened the pull request.

Commit:
Create HTML doc with employees and CSS

1) Create a basic HTML page containing drop-downs with employees and projects
2) Create a CSS doc with styles, applied to the HTML page

Commit:
Rename the folder and add JS doc

1) Renamed the folder to 'Website'
2) Added a JS doc with validation: employeesAvailability.js

Commit:
rename basic html doc to index.html

Alenapadsekina committed:
Change tabulation

Change tabulation to make the code easier to be read

Commit:
Change the "button" html code

Commit:
Merge branch 'Task_4-Employees_table' of https://github.com/Alenapadsekina/js_study into Task_4-Employees_table

Commit:
Fix the button position and style

1) Changed HTML doc: the button stands aside dropdowns now
2) Changed CSS doc: the button is blue and has a bigger font of the label

SergeBondar commented on Website/index.html:
```diff
@@ -0,0 +1,46 @@
+</head>
+
+<body>
+    <span class="teams">
```
Для вложенных структур обычно используют div

SergeBondar commented on Website/index.html:
```diff
@@ -0,0 +1,46 @@
+		</span>
+    	<span>
+			<select>
+				<option value="" class="dropdown"> </option>
```
class лучше повесить на select

SergeBondar commented on Website/index.html:
```diff
@@ -0,0 +1,46 @@
+		</span>
+	</span>
+	<span class="reportButton">
+		<input  class="button" type="button" value="Generate report" onclick="testButton(); employeesAvailability.js">
```
employeesAvailability.js - а вот это зачем тут?

Alenapadsekina committed:
HTML: updated 2nd dropdown; JS: added functions to autopopulate the 2nd dropdown

HTML doc:
1) changed href to src in script tag in the header
2) added onclick function to the options of the 1st dropdown
3) updated 2nd dropdown to be autopoopulated
4) changed the function name of the function, called by clicking 'Generate report' button
JS doc:
1) added basic employees array
2) added the function to choose a team in the 1st dropdown and populate the 2nd dropdown

Alenapadsekina committed:
add 'Generate report' button

HTML doc:
1) updated the section containing result table / error message
JS doc:
1) separated functions;
2) added function, creating a table
3) added validation and function, showing error messages
CSS doc:
1) added style to the table cells and header
2) added styes to error messages

SergeBondar commented on Website/employeesAvailability.js:
```diff
@@ -0,0 +1,160 @@
+    function defineTeamMembers(team) { // populate 'group' array with corresponding team members
+
+        for (var i = 0; i < employees.length; i++) {
+            if (employees[i].team == team) {
```
строгое

Alenapadsekina replied:
исправила

SergeBondar commented on Website/employeesAvailability.js:
```diff
@@ -0,0 +1,160 @@
+    var team = document.getElementById("teams").value;
+    var group = [];
+
+    function defineTeamMembers(team) { // populate 'group' array with corresponding team members
```
у тебя получилась очень сложная структура - часть функций объявлена глобально, часть внутри других функций
нужно унифицировать объявление функций

SergeBondar replied:
define, а в комментарии- populate?
давай еще раз это обсудим

SergeBondar commented on Website/employeesAvailability.js:
```diff
@@ -0,0 +1,160 @@
+
+            cleanDropdown();
+
+            function populateTeamMembers() { // create lines in 'Members' dropdown and popoulate them with group members
```
populate

Alenapadsekina replied:
исправила

SergeBondar commented on Website/employeesAvailability.js:
```diff
@@ -0,0 +1,160 @@
+                    var option = document.createElement("option");
+                    option.text = group[i].name;
+                    option.value = group[i].name;
+                    var select = document.getElementById("teamMembers");
```
это лучше объявить до цикла

Alenapadsekina replied:
исправила

SergeBondar commented on Website/employeesAvailability.js:
```diff
@@ -0,0 +1,160 @@
+            statusCell.classList.add("notAvailable");
+        }
+    }
+    var table = document.getElementById("reportTable");
```
это лучше сделать перед циклом, и тогда тебе не нужно будет каждый раз объявлять table
и еще - если ты сначала создашь строку, добавишь к ней всю инфу, и только потом добавишь ее в таблицу - dom обновится только один раз, а так - как у тебя - при каждой операции

SergeBondar commented on Website/employeesAvailability.js:
```diff
@@ -0,0 +1,160 @@
+
+var errorMessage;
+
+function validation(errorMessage) { // create error message
```
showError?

Alenapadsekina replied:
исправила

SergeBondar commented on Website/index.html:
```diff
@@ -0,0 +1,45 @@
+    	<span>
+    			<select id="teams" class="dropdown">
+					<option value="" > </option>
+					<option value="Main releases" class="dropdown" onclick="selectTeam()">Main releases</option>
```
onclick вешается на весь select один раз

Alenapadsekina replied:
исправила

SergeBondar reviewed (commented):
и что-то с отступами в html файле

Commit:
HTML and JS docs updates

HTML doc:
1) apply onclick to select (not every option)
JS doc:
1) place cleanDropdown and cleanOldResult inside other functions
2) replace '==' with '==='
3) rename function, that displays errors

SergeBondar commented on Website/employeesAvailability.js:
```diff
@@ -0,0 +1,159 @@
+
+        console.log(group);
+
+        function populateDropdown() {
```
Не совсем понимаю, зачем тебе эти 3 функции populateDropdown (странное название, ни о чем не говорит, cleanDropdown (обычно используется слово clear, но для 4 строчек редко создается отдельная функция) и populateTeamMembers
Почему бы просто не создать одну функцию - populateTeamMembers, и в ней реализовать всю вот эту логику?

SergeBondar commented on Website/employeesAvailability.js:
```diff
@@ -0,0 +1,159 @@
+
+    switch (team) {
+        case "":
+            errorMessage = "Please define a team";
```
а почему нельзя сразу строку передать в функцию?
зачем нам эта переменная?

SergeBondar commented on Website/index.html:
```diff
@@ -0,0 +1,48 @@
+			Teams
+		</span>
+    	<span>
+    			<select id="teams" class="dropdown" onclick="selectTeam()">
```
извини, для select тогда будет использоваться onchange событие

Commit:
Changed HTML and JS docs

JS doc:
1) simplified the function, populating Team Members dropdown
2) reduced qty of variables in generateReport function
HTML doc:
1) changed 'onclick' to 'onchange' in Teams select

Commit:
Updated HTML and JS docks

HTML doc:
1) added tabulation
JS doc:
1) put the function, defining team members, into a separate function

MaxMarasanov commented:
По условию, если член команды не выбран, то должна выводиться вся таблица, иначе - только для выбранного члена.

Сейчас же не выбрать члена команды нельзя в принципе (по умолчанию уже кто-то выбран), а при любом выборе строит таблицу для всей команды.

MaxMarasanov reviewed (changes requested):
Нужно подправить генерацию таблицы

Alenapadsekina committed:
report about 1 employee

added creating a report if only 1 employee is selected in 'Members' dropdown

Alenapadsekina committed:
added empty option
